src/plugins/data/public/ui/search_bar/lib/use_saved_query.ts

```diff
@@ -43,6 +43,8 @@ export const useSavedQuery = (props: UseSavedQueriesProps): UseSavedQueriesRetur
   // Handle saved queries
   const defaultLanguage = props.uiSettings.get('search:queryLanguage');
   const [savedQuery, setSavedQuery] = useState<SavedQuery | undefined>();
+  const prevSavedQuery = useRef<SavedQuery | undefined>();
+  prevSavedQuery.current = savedQuery;
```

Comment:
I think I don't understand why this `prevSavedQuery` is needed. 
It seems that `savedQuery` should be used directly. 

I guess when using `savedQuery`, then eslint rule asks to add it into deps? If so, should mute this rule for this case instead of working around linter with this additional code? Or am I missing the difference?

Reply:
Discussed offiline: 
1. Appeared that 
```
prevSavedQuery.current = savedQuery;
```
is not needed and removed. 

2. prevSavedQuery.current seems like also needed also to not disable eslint rule